packages/react/src/atoms/dividers/base.tsx

```diff
@@ -1,29 +1,47 @@
 import React, { ReactNode } from "react";
 import "../../../tailwind.css";
 import { useThemeContext } from "../../theme/ThemeProvider";
-import classNames from "../../utils/classNames";
+import resolvedDividerClassName from "../../utils/resolveDividerClassName";
+import resolvedStyleProps from "../../utils/resolvedStyleProps";
 
-export interface LabelBaseProps {
+export interface DividerBaseProps {
+  //String which will be displayed as divider text
   label: string;
-  htmlFor: string;
-  labelBaseClassNames?: any;
-  labelBaseStyles?: any;
+  //To create Varient of the Divider
+  variant?: string;
+  //Pass True if you want to display the label at the right of the divider
+  showAsSuffix?: boolean;
+  //Pass True if you want to display the label at the left of the divider
+  showAsPrefix?: boolean;
+  dividerBaseClasses?: {
+    //Margin for Label between the label
+    labelMargin?: any;
+    //Text color for label
+    textColor?: any;
+  };
 }
 
-const DividerBase = (props: DividerBaseProps) => {
+const DividerBase= (props:DividerBaseProps) => {
   const theme: any = useThemeContext();
-  return (
-    <label
-      htmlFor={props.htmlFor}
-      className={classNames(
-        theme.labelBaseClassNames,
-        props.labelBaseClassNames
-      )}
-      style={props.labelBaseStyles}
-    >
-      {props.label}
-    </label>
+
+  const dividerClassName = resolvedDividerClassName(props);
```

Comment:
@ShreyDhyani This part of enabling prefix and suffix, please see if we can do differently. Basically, you are styling differently if user is showing as prefix/suffix.

But I guess you can keep them again in theme only and use resolvedStyleProps.

In theme, you can add more items under dividerBaseClasses -> like centerClasses, suffixClasses, prefixClasses and then use

```
const suffixClasses = resolvedStyleProps(
    "dividerBaseClasses",
    [
      "labelMargin",
      "textColor"
    ],
    props,
    theme.dividerBaseClasses.suffixClasses
  );
```

The way it's done with resolvedDividerClassName, it doesn't give user option to override it via easily.